utils.go

```diff
@@ -9,6 +9,10 @@ import (
 	"github.com/google/go-querystring/query"
 )
 
+func ptr[T any](val T) *T {
```

Comment:
we do have some helper methods already in https://github.com/cloudflare/cloudflare-go/blob/master/convert_types.go that we're using in place of generics.